### ISSUE TITLE
Add frustum culling functionality for objects

### ISSUE THREAD

**Issue by lragnarsson** [lragnarsson/Emerald-Engine#11]

**Comment by lragnarsson** [comment]
I started working on frustum culling in the Labs, this is what I have done so far:

``` c
void calculateCullingFrustum(mat4 camMatrix) {
    float cLeft = -0.05f;
    float cRight = 0.05f;
    float cBottom = -0.05f;
    float cTop = 0.05f;
    float cNear = 0.2f;
    //float cFar = 50.f;

    vec3 topLeft = {cLeft, cTop, -cNear};
    vec3 bottomLeft = {cLeft, cBottom, -cNear};
    vec3 topRight = {cRight, cTop, -cNear};
    vec3 bottomRight = {cRight, cBottom, -cNear};
    vec3 farNormal = {0.f, 0.f, -1.f};

    mat4 invCam = InvertMat4(camMatrix);

    topLeft = MultVec3(invCam, topLeft);
    bottomLeft = MultVec3(invCam, topLeft);
    topRight = MultVec3(invCam, topLeft);
    bottomRight = MultVec3(invCam, topLeft);
    farNormal = MultVec3(invCam, farNormal);


    cullingFrustumNormals[0] = Normalize(CrossProduct(topLeft, bottomLeft)); // Left normal
    cullingFrustumNormals[1] = Normalize(CrossProduct(bottomRight, topRight)); // Right normal
    cullingFrustumNormals[2] = Normalize(CrossProduct(bottomLeft, bottomRight)); // Bottom normal
    cullingFrustumNormals[3] = Normalize(CrossProduct(topRight, topLeft)); // Top normal
    cullingFrustumNormals[4] = Normalize(farNormal); // Far normal

}

```

``` c
int sphereInFrustum(vec3 position, float radius) {
    int i;
    vec3 closestPoint;
    for (i=0; i < 5; i++) {
        closestPoint = VectorAdd(position, ScalarMult(cullingFrustumNormals[i], -radius));
        if (DotProduct(closestPoint, cullingFrustumNormals[i]) > 0.f)
            return 0;
    }
    return 1;
}
```

Might be used as a starting point!

**Comment by lragnarsson** [comment]
Ended up doing the culling in world coordinates. Merged into master with PR #17
